common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -192,9 +189,21 @@ def create_modulestore_instance(
         doc_store_config['read_preference'] = getattr(ReadPreference, doc_store_config['read_preference'])
 
     if disabled_xblocks:
-        disabled_xblock_types = [block.name for block in disabled_xblocks()]
+        def fetch_disabled_xblock_types():
+            """
+            Get the disabled xblock names, using the request_cache if possible to avoid hitting
+            a database every time the list is needed.
+            """
+            if request_cache:
+                if 'disabled_xblock_types' not in request_cache.data:
+                    request_cache.data['disabled_xblock_types'] = [block.name for block in disabled_xblocks()]
+                return request_cache.data['disabled_xblock_types']
+            else:
+                disabled_xblock_types = [block.name for block in disabled_xblocks()]
+
+            return disabled_xblock_types
     else:
-        disabled_xblock_types = ()
+        fetch_disabled_xblock_types = lambda: []
```

Comment:
Instead of using the existence of `disabled_xblocks` to conditionally define the function, could you please put the conditional check inside of  `fetch_disabled_xblock_types()` and return early if `disabled_xblocks` is `None`? What you're doing is totally valid, but it's a technique that's used seldom enough that it might confuse people who are newer to Python and are trying to debug this.